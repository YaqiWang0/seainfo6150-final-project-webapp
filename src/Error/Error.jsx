import React from 'react'
import {Link} from "react-router-dom"
import style from './NotFoundPage.module.css'

const Error = () => {
    return (
        <div className={style.container}>
            <div className={style.picBlock}>
                <div className={style.pic}></div>
            </div>
            <div className={style.textContainer}>
                <div className={style.textBlock}>
                    <div className={style.text}>
                        404
                    </div>
                    <div className={style.message}>
                        Page Not Found
                    </div>
                    <div className={style.content}>The page you are looking for does not exist. How you got here is a mystery. But you can click the button below to go back to the homepage.</div>
                    <button className={style.btn}><Link to="/">HOME</Link></button>
                </div>
            </div>
        </div>
    )
}

export default Error
