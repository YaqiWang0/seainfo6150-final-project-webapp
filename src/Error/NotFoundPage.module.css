.container {
    width: 100%;
    height: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
}

.text {
    font-size: 15vw;
    font-weight: bold;
    color: white;
    text-align: center;
}

.message {
    font-size: 7vw;
    font-weight: bold;
    text-align: center;
    color: white;
}