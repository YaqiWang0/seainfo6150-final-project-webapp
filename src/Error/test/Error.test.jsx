import React from "react";

import Error from "../Error";

describe("Error test", () => {

    it('render successfully', function () {
        const {container} = render(
            <Error/>
        );
        expect(container).toMatchSnapshot();
    })
})