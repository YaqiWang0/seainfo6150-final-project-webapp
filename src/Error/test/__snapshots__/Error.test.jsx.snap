// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error test render successfully 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="picBlock"
    >
      <div
        class="pic"
      />
    </div>
    <div
      class="textContainer"
    >
      <div
        class="textBlock"
      >
        <div
          class="text"
        >
          404
        </div>
        <div
          class="message"
        >
          Page Not Found
        </div>
        <div
          class="content"
        >
          The page you are looking for does not exist. How you got here is a mystery. But you can click the button below to go back to the homepage.
        </div>
        <button
          class="btn"
        >
          <a
            href="/"
          >
            HOME
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Error test when the item is not clicked 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="picBlock"
    >
      <div
        class="pic"
      />
    </div>
    <div
      class="textContainer"
    >
      <div
        class="textBlock"
      >
        <div
          class="text"
        >
          404
        </div>
        <div
          class="message"
        >
          Page Not Found
        </div>
        <div
          class="content"
        >
          The page you are looking for does not exist. How you got here is a mystery. But you can click the button below to go back to the homepage.
        </div>
        <button
          class="btn"
        >
          <a
            href="/"
          >
            HOME
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
